internal/services/extendedlocation/extended_location_custom_location.go

```diff
@@ -0,0 +1,348 @@
+			}
+
+			customLocationProps := customlocations.CustomLocationProperties{}
+
+			if model.Authentication != nil {
+				auth := model.Authentication[0]
+				customLocationProps.Authentication = &customlocations.CustomLocationPropertiesAuthentication{
+					Type:  &auth.Type,
+					Value: &auth.Value,
+				}
+			}
+
+			if model.ClusterExtensionIds != nil {
+				customLocationProps.ClusterExtensionIds = &model.ClusterExtensionIds
+			}
+
+			if model.DisplayName != "" {
+				customLocationProps.DisplayName = &model.DisplayName
+			}
+
+			if model.HostResourceId != "" {
+				customLocationProps.HostResourceId = &model.HostResourceId
+			}
+
+			if model.HostType != "" {
+				hostType := customlocations.HostType(model.HostType)
+				customLocationProps.HostType = &hostType
+			}
+
+			if model.Namespace != "" {
+				customLocationProps.Namespace = &model.Namespace
+			}
+
+			props := customlocations.CustomLocation{
+				Id:         utils.String(id.ID()),
+				Location:   model.Location,
+				Name:       utils.String(model.Name),
+				Properties: &customLocationProps,
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, props); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r CustomLocationResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ExtendedLocation.CustomLocationsClient
+			id, err := customlocations.ParseCustomLocationID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("reading %s: %+v", *id, err)
+			}
+
+			if model := resp.Model; model != nil {
+				props := model.Properties
+
+				state := CustomLocationResourceModel{
+					Name:              id.CustomLocationName,
+					ResourceGroupName: id.ResourceGroupName,
+					Location:          model.Location,
+				}
+
+				if props != nil && props.Authentication != nil {
+					authType := pointer.From(props.Authentication.Type)
+					authValue := pointer.From(props.Authentication.Value)
+					state.Authentication = []AuthModel{
+						{
+							Type:  authType,
+							Value: authValue,
+						},
+					}
+				}
+
+				if props.ClusterExtensionIds != nil {
+					state.ClusterExtensionIds = *props.ClusterExtensionIds
+				}
+
+				if props.DisplayName != nil {
+					state.DisplayName = *props.DisplayName
+				}
+
+				if props.HostResourceId != nil {
+					state.HostResourceId = *props.HostResourceId
+				}
+
+				if props.HostType != nil {
+					state.HostType = string(*props.HostType)
+				}
+
+				if props.Namespace != nil {
+					state.Namespace = *props.Namespace
+				}
+
+				return metadata.Encode(&state)
+			}
```

Comment:
Could we update this to use `pointer.From`, it could simplify the codes because it has a nil-check inside.

Reply:
Will do.